src/Monolog/Handler/SocketHandler.php

```diff
@@ -236,8 +236,31 @@ public function getChunkSize(): ?int
      */
     public function isConnected(): bool
     {
-        return is_resource($this->resource)
-            && !feof($this->resource);  // on TCP - other party can close connection.
+        /*
+         * We try to probe that we are not connected in order to return fast.
+         */
+        if (!is_resource($this->resource)) {
+            return false;
+        }
+
+        if (feof($this->resource)) {
+            return false;
+        }
+
+        $info = stream_get_meta_data($this->resource);
+
+        if (isset($info['timed_out']) && $info['timed_out']) {
+            return false;
+        }
```

Comment:
This check is already covered below when writing IMO so not sure this is needed.